pkg/services/govmomi/service.go

```diff
@@ -224,6 +224,15 @@ func (vms *VMService) reconcileMetadata(ctx *virtualMachineContext) (bool, error
 		return false, err
 	}
 
+	// checking if any of the network devices for the VSphereVM was removed
+	if statuses := len(ctx.State.Network); statuses > 0 {
+		if statuses > len(ctx.VSphereVM.Spec.Network.Devices) {
```

Comment:
The network statuses on the VM show the current number of devices attached whereas the `vm.spec.network.devices` holds the expected number of devices.
If a device is removed, the `statuses` list gets updated and will be lesser than the expected number of devices. Hence this condition should be flipped to
```golang
if statuses < len(ctx.VSphereVM.Spec.Network.Devices)
```

Also, have a question around, does it make sense to remove the check at line 228. Would that be necessary, or am I missing a case.

cc @yastij

Reply:
let's remove this check since folks are still managing network devices outside of CAPV.